sdks/python/apache_beam/io/filesystem.py

```diff
@@ -166,6 +177,9 @@ def _initialize_decompressor(self):
       self._decompressor = bz2.BZ2Decompressor()
     elif self._compression_type == CompressionTypes.DEFLATE:
       self._decompressor = zlib.decompressobj()
+    elif self._compression_type == CompressionTypes.ZSTD:
+      self._decompressor = zstandard.ZstdDecompressor(
+          max_window_size=2147483648).decompressobj()
```

Comment:
I know this seems arbitrary (and probably is), but without this param the decompressor fails with error: 
```
zstd.ZstdError: zstd decompress error: Frame requires too much memory for decoding
```
I thought about using `DEFAULT_READ_BUFFER_SIZE` but it seems too low for this param. I can either create a constant or a new parameter if necessary, let me know.

Reply:
We should have some explanation here in a comment.
Is there a related bug/discussion with zstd  maintainers that gives this recommendation from an authoritative source?

Reply:
thank you for the suggestion, I agree and found this: https://github.com/indygreg/python-zstandard/issues/157
apparently it is related to the compression level used. What do you think about adding this as a comment? It doesn't seem like the library intends to fix this, as the issue was closed, but I still think it's better to leave the value as it seems to work as a general recommendation (for my use case I tested in 10GB+ compressed or 100GB+ decompressed files and it is working fine)

Reply:
thank you for the suggestion, I agree and found this: https://github.com/indygreg/python-zstandard/issues/157
apparently it is related to the compression level used. What do you think about referencing this issue as a comment? It doesn't seem like the library intends to fix this in an automated way, as the issue was closed, but I still think it's better to leave the value as it seems to work as a general recommendation (for my use case I tested in 10GB+ compressed or 100GB+ decompressed files in several difference sizes and workers from 5GB to 64GB RAM and it is working fine)

Reply:
sure. let's do that.

Reply:
can we also add a unit test exercising this compression on a small payload?

Reply:
@tvalentyn done, for the tests I reused the filesystem_test test cases for zst, hopefully that is enough, but let me know